### PR TITLE
Endpoint to reset password

### PR DESCRIPTION
**Pull request by SilviaPabon**
- 2 new fields in `users` collection: `resetPassCode` y`resetPassCodeExp` (time expiration) are included (the idea is in a future PR, create a new collection with fields like `resetPassCode` or `validationCode`)
- A code is created and added in `resetPassCode` when user request reset the password and its sended to the mail. To prove this, make a POST request to `/reset_password` including in the body the email that needs reset the password.
- Prove making a PUT request to `/reset_password`, to change password:
`{
	"email": "youremail@mail.com",
	"resetPassCode": "######",
	"password": "NewPassword"
}`

### CODE REVIEW

**Review by andres123dbh — APPROVED** [review]

File: api/routes/router.go [hunk 1]
```diff
@@ -11,6 +11,8 @@ func SetupRoutes(engine *gin.Engine) {
 	engine.POST("/signup", controllers.HandleSignUp)
 	engine.POST("/code_validation", controllers.HandleCodeValidation)
 	engine.POST("/new_code", controllers.HandleNewCodeValidation)
+	engine.POST("/reset_password", controllers.HandleCodeResetPassword)
+	engine.PUT("/reset_password", controllers.HandleResetPassword)
```
Comment (PedroChaparro) [inline api/routes/router.go:15]: What do you think about: 

`/user/password/code` for the POST route and
`/user/password` for the PUT route.

File: api/controllers/user.go [hunk 2]
```diff
@@ -184,3 +184,104 @@ func HandleNewCodeValidation(c *gin.Context) {
...
+		c.AbortWithStatusJSON(http.StatusNotFound, gin.H{"message": "This Email has not been registered"})
+		return
+	}
+	//generate code
+	resetPasswordCode := utils.GetValidationCode()
+
+	//update in database reset password code
+	err = models.UpdateResetPassCode(form.Email, resetPasswordCode)
+	if err != nil {
+		fmt.Println(err)
```
Comment (PedroChaparro) [inline api/controllers/user.go:213]: Please, remove / comment those debug messages when the tasks is fully completed.

File: api/controllers/user.go [hunk 3]
```diff
@@ -184,3 +184,104 @@ func HandleNewCodeValidation(c *gin.Context) {
...
+	}
+
+	//send mail with code to help reset password
+	err = utils.SendEmail(form.Email, "Here is your validation code, to reset your password", resetPasswordCode)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{"message": "Internal server error"})
+		return
+	}
+
+	c.IndentedJSON(http.StatusOK, gin.H{"message": "New Code, to reset password, created and sended"})
```
Comment (PedroChaparro) [inline api/controllers/user.go:225]: Responses should follow this structure: 

```json
{
  "error": "boolean",
  "message": "String", 
  ... 
}
```

Please add the `error` field, at this way we can start standardize our API.

**Review by PedroChaparro — CHANGES_REQUESTED** [review]

**Review by PedroChaparro — APPROVED** [review]
**Note to myself:** It's recommended to use camelCase to name JSON fields.

**Comment by Woynert** [comment]
Please see #78

**Pull request merged**